presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftMetastoreUtil.java

```diff
@@ -570,6 +572,18 @@ public static OptionalLong fromMetastoreNullsCount(long nullsCount)
         return OptionalLong.of(nullsCount);
     }
 
+    /**
+     * Impala 'COMPUTE STATS' write 1 as the numTrue and -1 as the numFalse
+     * @see <a href="https://github.com/apache/impala/blob/0cbe37afc8f77fd8e0aef7e710e2d1b214c1c5b1/fe/src/main/java/org/apache/impala/catalog/ColumnStats.java#L269">Impala ColumnStats.java#L269</>
+     */
+    public static OptionalLong fromMetastoreNumFalse(long numFalse)
+    {
+        if (numFalse == -1L) {
```

Comment:
The problem here is that, when you see `-1` you know that the value is incorrect, but when you see `1` you don't know this. This makes me thing that number of `false` and `true` values in hive statistics (when Impala is used) is not reliable and should not be used at all.

As I understand the code,`com.facebook.presto.hive.metastore.BooleanStatistics#getTrueCount` and `com.facebook.presto.hive.metastore.BooleanStatistics#getFalseCount` are only used to calculate NDV Presto statistic. Hence I see two options here:
1. either return `1` when `numFalse = -1`  (assuming there were some `false` values)
2. not use `numFalse` and `numTrue` hive stats at all, and assume `NDV=2` for boolean columns

I prefer option number 1.

@sopel39 @findepi @electrum Any thoughts on this?

Reply:
> * either return `1` when `numFalse = -1`  (assuming there were some `false` values)

This will work only until we actually use "num trues" in the engine. Let's not place this trap.

> 2\. not use `numFalse` and `numTrue` hive stats at all, and assume `NDV=2` for boolean columns

When these stats actually matter, we may be 50% off when we don't get them right. And boolean is likely used for flags like "is-valid-account" where assumption of uniform distribution is not going to get us anywhere.

There is also option 3: _When `numFalse` and `numTrue` in Hive stats are valid values, use them. When any of them is -1, return unknown for both of them._
This is good approach for all stats-writers we discussed so far (Presto, Hive, Impala). In the future, when we come along a new stats-writer that violates these assumptions, we can revisit.